Check for ANGLE-supported formats not available in WebGL

lexaknyazev opened the pull request:
#### 3728a98461ce1ad0e74b7a416e823bd17ae6872e
<pre>
Check for ANGLE-supported formats not available in WebGL
<a href="https://bugs.webkit.org/show_bug.cgi?id=223352">https://bugs.webkit.org/show_bug.cgi?id=223352</a>

Reviewed by Kimmo Kinnunen.

Some internal formats are supported by ANGLE implicitly
or enabled by WebKit explicitly, although never being
exposed to WebGL apps.

Fixes remaining failures in
* conformance2/state/gl-object-get-calls.html

* LayoutTests/TestExpectations:
* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp:
(WebCore::WebGL2RenderingContext::getInternalformatParameter):
* Source/WebCore/platform/graphics/GraphicsContextGL.h:

Canonical link: <a href="https://commits.webkit.org/253286@main">https://commits.webkit.org/253286@main</a>
</pre>

kkinnunen-apple commented on Source/WebCore/html/canvas/WebGL2RenderingContext.cpp:
```diff
@@ -803,6 +803,16 @@ WebGLAny WebGL2RenderingContext::getInternalformatParameter(GCGLenum target, GCG
 
+    // These formats are never exposed to WebGL apps but may be accepted by ANGLE.
+    switch (internalformat) {
+    case 0x6AC0: // GL_BGRA8_SRGB_ANGLEX
```
Instead of numeric constants in the code, could you add these to the GraphicsContextGL enum like the rest of them?

lexaknyazev committed:
Check for ANGLE-supported formats not available in WebGL
https://bugs.webkit.org/show_bug.cgi?id=223352

Reviewed by Kimmo Kinnunen.

Some internal formats are supported by ANGLE implicitly
or enabled by WebKit explicitly, although never being
exposed to WebGL apps.

Fixes remaining failures in
* conformance2/state/gl-object-get-calls.html

* LayoutTests/TestExpectations:
* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp:
(WebCore::WebGL2RenderingContext::getInternalformatParameter):
* Source/WebCore/platform/graphics/GraphicsContextGL.h:

Canonical link: https://commits.webkit.org/253286@main
